indico/modules/rb/models/reservations.py

```diff
@@ -91,7 +92,7 @@ def convert_legacy_repeatability(cls, repeat):
             raise KeyError('Undefined old repeat: {}'.format(repeat))
 
 
-class Reservation(Serializer, db.Model):
+class Reservation(Serializer, LinkMixin, db.Model):
```

Comment:
What about keeping all the link information in a separate model, something like `ReservationLink`, that has an 1:1 relationship to `Reservation.?
That way we keep stuff we don't need very often out of the Reservation model and there aren't any drawbacks except that linking is slightly more complicated (need to create the ReservationLink if it doesn't exist yet).